lib/jwk/keystore.js

```diff
@@ -395,6 +395,11 @@ var JWKStore = function(registry, parent) {
     value: function(props, local) {
       props = props || {};
 
+      // workaround for issues/109
+      if (typeof props.kid === "number") {
```

Comment:
as ugly as that looks, that looks the most correct to me.  I'm more comfortable with that than simply checking for number.